### PR TITLE
[OSPP]: Personalized LLM Agent based on KubeEdge-Ianvs Cloud-Edge Collaboration

### PR DESCRIPTION
**Pull request by Frank-lilinjie**
The PR of _Personalized LLM Agent based on KubeEdge-Ianvs Cloud-Edge Collaboration_

### CODE REVIEW

**Review by MooreZheng — CHANGES_REQUESTED** [review]
There are CI issues that needs to be resolved
`Run pylint '/home/runner/work/ianvs/ianvs/core'
************* Module core.testcasecontroller.algorithm.paradigm.singletask_learning.singletask_learning_active_boost
core/testcasecontroller/algorithm/paradigm/singletask_learning/singletask_learning_active_boost.py:66:4: R09[1](https://github.com/kubeedge/ianvs/actions/runs/11092223886/job/30816995965?pr=154#step:5:1)7: Too many positional arguments (7/5) (too-many-positional-arguments)
************* Module core.testenvmanager.dataset.dataset
core/testenvmanager/dataset/dataset.py:119:4: R091[7](https://github.com/kubeedge/ianvs/actions/runs/11092223886/job/30816995965?pr=154#step:5:8): Too many positional arguments (8/5) (too-many-positional-arguments)
core/testenvmanager/dataset/dataset.py:206:4: R0917: Too many positional arguments (6/5) (too-many-positional-arguments)
core/testenvmanager/dataset/dataset.py:213:4: R0917: Too many positional arguments (7/5) (too-many-positional-arguments)
core/testenvmanager/dataset/dataset.py:246:4: R0917: Too many positional arguments (7/5) (too-many-positional-arguments)
core/testenvmanager/dataset/dataset.py:2[8](https://github.com/kubeedge/ianvs/actions/runs/11092223886/job/30816995965?pr=154#step:5:9)5:4: R0917: Too many positional arguments (7/5) (too-many-positional-arguments)
core/testenvmanager/dataset/dataset.py:329:4: R0917: Too many positional arguments (7/5) (too-many-positional-arguments)
core/testenvmanager/dataset/dataset.py:368:4: R0[9](https://github.com/kubeedge/ianvs/actions/runs/11092223886/job/30816995965?pr=154#step:5:10)17: Too many positional arguments (6/5) (too-many-positional-arguments)

-----------------------------------
Your code has been rated at 9.95/[10](https://github.com/kubeedge/ianvs/actions/runs/11092223886/job/30816995965?pr=154#step:5:11)

Error: Process completed with exit code 8.`

**Review by hsj576 — APPROVED** [review]
/lgtm

File: examples/llm-agent/singletask_learning_bench/testalgorithms/basemodel.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    def load(self, model_url, **kwargs):
+        if model_url:
+            print("load model url: ",model_url)
+
+    def save(self, model_path = None):
+        pass
+
+    def load_json(self, data_dir, tokenizer, token_factor = 32):
+        MYjson=datasets.load_dataset("json",data_files=data_dir) # 加载Json数据集
```
Comment (MooreZheng) [inline examples/llm-agent/singletask_learning_bench/testalgorithms/basemodel.py:78]: For load_json of basemodel.py, need to translate Chinese comments into English to facilitate international impacts

File: examples/llm-agent/singletask_learning_bench/testalgorithms/basemodel.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        # train_data=self.preprocess(train_data, self.MAX_LENGTH, tokenizer)
+        ds=MYjson.map(self.preprocess,fn_kwargs={"MAX_LENGTH":self.MAX_LENGTH,"tokenizer":tokenizer},batched=True,batch_size=2,remove_columns=['role','content'])
+
+        filtered_ds=ds.filter(lambda example:not None in example["labels"]) # 过滤掉标签为 None的样本
+        return filtered_ds
+    
+
+    def preprocess(self, samples, MAX_LENGTH, tokenizer):
+        input_ids,attention_mask,labels=[],[],[] # 初始化三个空列表
```
Comment (MooreZheng) [inline examples/llm-agent/singletask_learning_bench/testalgorithms/basemodel.py:87]: For preprocess of basemodel.py, need to translate Chinese comments into English

**Review by MooreZheng — CHANGES_REQUESTED** [review]
To facilitate international impacts, need to translate Chinese comments into English for load_json and preprocess of basemodel.py.

**Review by MooreZheng — CHANGES_REQUESTED** [review]
Just to remind @Frank-lilinjie : Chinese statements need to be refined into English before merging. 

Besides, this pull request contains 15 commits, which might make maintenance difficult, considering the number of contributors, pull requests, and their commits in KubeEdge Ianvs recently. In the final stage, @Frank-lilinjie can squash the commits into one using [rebase techniques](https://blog.kinboy.wang/2019/04/09/Git-%E5%8E%8B%E7%BC%A9%E5%A4%9A%E4%B8%AAcommit%E4%B8%BA%E5%8D%95%E4%B8%AAcommit/).

**Comment by Frank-lilinjie** [comment]
> Just to remind @Frank-lilinjie : Chinese statements need to be refined into English before merging.
> 
> Besides, this pull request contains 15 commits, which might make maintenance difficult, considering the number of contributors, pull requests, and their commits in KubeEdge Ianvs recently. In the final stage, @Frank-lilinjie can squash the commits into one using [rebase techniques](https://blog.kinboy.wang/2019/04/09/Git-%E5%8E%8B%E7%BC%A9%E5%A4%9A%E4%B8%AAcommit%E4%B8%BA%E5%8D%95%E4%B8%AAcommit/).

copy that

File: examples/llm-agent/singletask_learning_bench/testenv/rouge.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    rouge=evaluate.load('./examples/LLM-Agent-Benchmark/evaluate/metrics/rouge')
+    y_prednew=[]
+    for i in range(len(y_pred)):
+        y_prednew.append(y_pred[i]["generated_text"])
+    rou_score = rouge.compute(predictions = y_prednew, references=y_true, use_aggregator=True)
+    rougeL = rou_score['rougeL'] * 10
+    return rougeL
+
+def calculate_mean(lst):
+    print(lst)
```
Comment (hsj576) [inline examples/llm-agent/singletask_learning_bench/testenv/rouge.py:37]: The "print" should be removed or replaced by "LOGGER.info".

**Review by hsj576 — CHANGES_REQUESTED** [review]
The "print" should be removed or replaced by "LOGGER.info".

File: examples/llm-agent/singletask_learning_bench/testenv/rouge.py (hunk 3 above)
Comment (Frank-lilinjie) [reply examples/llm-agent/singletask_learning_bench/testenv/rouge.py:37]: copy that

**Review by hsj576 — APPROVED** [review]
/lgtm

**Review by MooreZheng — CHANGES_REQUESTED** [review]
This pull request contains 20 commits, which might make maintenance difficult, considering the number of contributors, pull requests, and their commits in KubeEdge Ianvs recently. In the final stage, @Frank-lilinjie can squash the commits into one using [rebase techniques](https://blog.kinboy.wang/2019/04/09/Git-%E5%8E%8B%E7%BC%A9%E5%A4%9A%E4%B8%AAcommit%E4%B8%BA%E5%8D%95%E4%B8%AAcommit/).

**Review by MooreZheng — APPROVED** [review]
All concerns are tackled. Well done! @Frank-lilinjie

**Pull request merged**